src/mono/wasm/debugger/BrowserDebugProxy/FirefoxMonoProxy.cs

```diff
@@ -0,0 +1,910 @@
+                            if (messageArgs != null && messageArgs.Count == 2)
+                            {
+                                if (messageArgs[0].Value<string>() == MonoConstants.RUNTIME_IS_READY && messageArgs[1].Value<string>() == MonoConstants.RUNTIME_IS_READY_ID)
+                                {
+                                    await OnDefaultContext(sessionId, new ExecutionContext(new MonoSDBHelper (this, logger, sessionId), 0, actorName), token);
+                                    await RuntimeReady(sessionId, token);
+                                }
+                            }
+                        }
+                        break;
+                    }
+                case "target-available-form":
+                    {
+                        actorName = args["target"]["consoleActor"].Value<string>();
+                        break;
+                    }
+            }
+            return false;
+        }
+
+        //from ide
+        protected override async Task<bool> AcceptCommand(MessageId sessionId, JObject args, CancellationToken token)
+        {
+            if (args["type"] == null)
+                return await Task.FromResult(false);
+
+            switch (args["type"].Value<string>())
+            {
+                case "resume":
+                    {
+                        if (!contexts.TryGetValue(sessionId, out ExecutionContext context))
+                            return false;
+                        if (context.CallStack == null)
+                            return false;
+                        if (args["resumeLimit"] == null || args["resumeLimit"].Type == JTokenType.Null)
+                        {
+                            await OnResume(sessionId, token);
+                            return false;
+                        }
+                        switch (args["resumeLimit"]["type"].Value<string>())
+                        {
+                            case "next":
+                                await context.SdbAgent.Step(context.ThreadId, StepKind.Over, token);
+                                break;
+                            case "finish":
+                                await context.SdbAgent.Step(context.ThreadId, StepKind.Out, token);
+                                break;
+                            case "step":
+                                await context.SdbAgent.Step(context.ThreadId, StepKind.Into, token);
+                                break;
+                        }
+                        await SendResume(sessionId, token);
+                        return true;
+                    }
+                case "attach":
+                    {
+                        threadName = args["to"].Value<string>();
+                        break;
+                    }
+                case "source":
+                    {
+                        if (args["to"].Value<string>().StartsWith("dotnet://"))
```

Comment:
This check isn't needed, as `OnGetScriptSource` will check as the first thing.

So, this can be similar to MonoProxy:
```
                        string script = args?["scriptId"]?.Value<string>();
                        return await OnGetScriptSource(id, script, token);
```